tests/expectations.json

```diff
@@ -8754,53 +8754,7 @@
   "built-ins/TypedArray/prototype/byteLength/resized-out-of-bounds-2.js": "CRASH",
   "built-ins/TypedArray/prototype/byteOffset/resized-out-of-bounds.js": "CRASH",
   "built-ins/TypedArray/prototype/entries/resizable-buffer.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/BigInt/coerced-indexes.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/BigInt/detached-buffer.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/BigInt/fill-values-conversion-once.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/BigInt/fill-values-custom-start-and-end.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/BigInt/fill-values-non-numeric-throw.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/BigInt/fill-values-non-numeric.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/BigInt/fill-values-relative-end.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/BigInt/fill-values-relative-start.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/BigInt/fill-values-symbol-throws.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/BigInt/fill-values.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/BigInt/get-length-ignores-length-prop.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/BigInt/return-abrupt-from-end-as-symbol.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/BigInt/return-abrupt-from-end.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/BigInt/return-abrupt-from-set-value.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/BigInt/return-abrupt-from-start-as-symbol.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/BigInt/return-abrupt-from-start.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/BigInt/return-abrupt-from-this-out-of-bounds.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/BigInt/return-this.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/absent-indices-computed-from-initial-length.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/coerced-end-detach.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/coerced-indexes.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/coerced-start-detach.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/coerced-value-detach.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/coerced-value-start-end-resize.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/detached-buffer.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/fill-values-conversion-once.js": "CRASH",
-  "built-ins/TypedArray/prototype/fill/fill-values-conversion-operations-consistent-nan.js": "CRASH",
   "built-ins/TypedArray/prototype/fill/fill-values-conversion-operations.js": "CRASH",
```

Comment:
In this test, it seems that when 2147483647 is passed to fill, it needs to become 2147483648, but I’m wondering if the issue might be with the use of v.into_value()… 👀

https://github.com/trynova/nova/blob/115c527d70ed368b12ee2d90598e1aead25f72c5/nova_vm/src/ecmascript/builtins/indexed_collections/typed_array_objects/typed_array_intrinsic_object.rs#L1171-L1447

Reply:
Alright; us using SmallInteger both for safe integers and small bigints has bit us in the ass here. We're converting into a number but we use the SmallInteger conversion which effectively tries to convert into a small bigint. That conversion succeeds, even though a safe integer conversion wouldn't.

Reply:
So that’s what it was all about…

Reply:
Alright, I spawned #613 based on this.